app/controllers/posts_controller.rb

```diff
@@ -9,13 +9,13 @@ def show
   end
 
   def new
-    @post = Post.new(author: current_user)
-    puts @post.author.name
+    @post = Post.new(user: current_user)
+    puts @post.user.name
```

Comment:
- [ ] `@post = Post.new` is enough for here.
- [ ] we dont require `def like & def comments` as we are creating like in LikesController, & CommentsController. Kindly remove line 34-51.& create a new controller LikesController & CommentsController with method `new & create` method to create new like & comment
- Make sure you create routes for this new controllers.
  ```
   POST   /posts/:post_id/comments(.:format)     comments#create
  GET   /posts/:post_id/comments/new(.:format)     comments#new
  POST /posts/:post_id/likes(.:format)                      likes#create

```